News-Api/src/Articles.jsx

```diff
@@ -1,48 +1,50 @@
 import { getArticles } from "./api.js";
-import { useState, useEffect, useContext} from "react";
-import { Link } from "react-router-dom";
-import ArticleListCard from './ArticleListCard.jsx';
+import { useState, useEffect } from "react";
+import { Link, useParams } from "react-router-dom";
+import ArticleListCard from "./ArticleListCard.jsx";
 import Header from "./Header.jsx";
 import { dateFormatter } from "./dateUtils.js";
 
-
 const Home = () => {
-
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-
+  const [isError, setIsError] = useState(false);
+  const { topic } = useParams();
 
   useEffect(() => {
-    getArticles().then((articleData) => {
-     
-          setArticles(articleData);
-
-      setIsLoading(false);
-    });
-  }, []);
+    getArticles(topic)
+      .then((articleData) => {
+        setIsLoading(true);
+        setArticles(articleData);
+        setIsLoading(false);
```

Comment:
You've got both setting the loading to true and false in the same .then
Think you mean to have this on the otherside of the getArticles